### PR TITLE
feat: add riscv64 support & fix minor bugs

### PR DESCRIPTION
**Pull request by RevySR**
Add riscv64 support patch.
Fix loong64 patch.
Fix gcc-12 build error.

Signed-off-by: Han Gao <gaohan@iscas.ac.cn>

### CODE REVIEW

**Comment by imaiguo** [comment]
Why change this file [debian/patches/uniontech0004-fit-loong64.patch]

File: debian/patches/uniontech0004-fit-loong64.patch [hunk 1]
```diff
@@ -8152,8 +8143,8 @@ index db16f8ccee..8340b37986 100644
    #undef  LSS_RETURN
    #if (defined(__i386__) || defined(__x86_64__) || defined(__ARM_ARCH_3__) \
 -       || defined(__ARM_EABI__) || defined(__aarch64__) || defined(__s390__))
-+       || defined(__ARM_EABI__) || defined(__aarch64__) || defined(__s390__)) \
```
Comment (RevySR) [inline debian/patches/uniontech0004-fit-loong64.patch]: @imaiguo 这里的宏括号写错位置了 然后修改后用quilt refresh刷新补丁就这样了

Comment (imaiguo) [reply debian/patches/uniontech0004-fit-loong64.patch]: 建议重新更新一下，不要过度依赖命令行自动化工具

**Pull request closed without merging**